### PR TITLE
Adding SSE for groupchats

### PR DESCRIPTION
**Pull request by ferigis**
Adding group messages to the SSE connection

### CODE REVIEW

File: test/ejabberd_tests/tests/rest_client_SUITE.erl [hunk 1]
```diff
@@ -87,6 +88,24 @@ msg_is_sent_and_delivered_over_sse(ConfigIn) ->
 
     stop_sse(Conn).
 
+msg_is_sent_and_delivered_over_sse_muc(Config) ->
```
Comment (michalwski) [inline test/ejabberd_tests/tests/rest_client_SUITE.erl]: How about messages with notifications about users joining/leaving the room? I think a testcase for that would be useful.

File: test/ejabberd_tests/tests/rest_client_SUITE.erl [hunk 2]
```diff
@@ -87,6 +88,24 @@ msg_is_sent_and_delivered_over_sse(ConfigIn) ->
...
+    escalus:fresh_story(Config, [{alice, 1}, {bob, 1}], fun(Alice, Bob) ->
```
Comment (michalwski) [inline test/ejabberd_tests/tests/rest_client_SUITE.erl]: Does `Bob` get the messages via XMPP as well? or only SSE?

File: test/ejabberd_tests/tests/rest_client_SUITE.erl [hunk 3]
```diff
@@ -162,7 +199,8 @@ user_is_invited_to_a_room(Config) ->
         RoomInfo = get_room_info({alice, Alice}, RoomID),
         true = is_participant(Bob, <<"member">>, RoomInfo),
         IQ = escalus_stanza:iq_get(<<"urn:xmpp:muclight:0#affiliations">>, []),
-        RoomJID = <<RoomID/binary, "@muclight.localhost">>,
+        Host = ct:get_config({hosts, mim, domain}),
+        RoomJID = <<RoomID/binary, "@muclight.",Host/binary>>,
```
Comment (gadgetci) [inline test/ejabberd_tests/tests/rest_client_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 203

File: test/ejabberd_tests/tests/rest_client_SUITE.erl [hunk 4]
```diff
@@ -87,6 +89,41 @@ msg_is_sent_and_delivered_over_sse(ConfigIn) ->
...
+    Alice = escalus_users:get_userspec(Config, alice),
+    RoomID = given_new_room({alice, Alice}),
+    Conn = connect_to_sse({bob, Bob}),
+    given_user_invited({alice, Alice}, RoomID, Bob),
+    Event = wait_for_event(Conn),
+    Data = jiffy:decode(maps:get(data, Event), [return_maps]),
+    ct:pal("~p", [Data]),
+    BobJID = user_jid(Bob),
+    Host = ct:get_config({hosts, mim, domain}),
+    RoomJID = <<RoomID/binary, "@muclight.",Host/binary>>,
```
Comment (gadgetci) [inline test/ejabberd_tests/tests/rest_client_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 119

File: doc/http-api/client_swagger.yml [hunk 5]
```diff
@@ -218,23 +218,44 @@ paths:
...
+           * messages - a message sent from participant to the room
+             ```JSON
+             {"type":"message",
+              "timestamp":1483609513435964,
+              "room":"1483-609513-293487",
+              "id":"9746c6d5-e014-4391-b559-b5581ee8e43b",
+              "from":"alice@wonderland.lit",
+              "body":"Hi all!"}
+             ```
+           * notifications - a message sent by the room when a new participant subscribes or leaves the room
```
Comment (Nyco) [inline doc/http-api/client_swagger.yml]: How about we name that "meta" or "event" but not "message"?

Comment (michalwski) [inline doc/http-api/client_swagger.yml]: Good idea :)

**Pull request merged**